src/DDTrace/OpenTelemetry/SpanBuilder.php

```diff
@@ -92,6 +96,38 @@ public function addLink(SpanContextInterface $context, iterable $attributes = []
         return $this;
     }
 
+    public function addEvent(string $name, int $timestamp = null, iterable $attributes = []): SpanBuilderInterface
+    {
+        $attributesArray = Attributes::create($attributes);
+        $nanoTimestamp = ($timestamp !== null && $timestamp < 1e9) ? $timestamp * 1000 : ($timestamp ?? (int)(microtime(true) * 1e9)); // Convert microseconds to nanoseconds if needed
+        $this->events[] = new Event($name, $nanoTimestamp, Attributes::create($attributes));
+        return $this;
+    }
+
+    public function recordException(Throwable $exception, iterable $attributes = []): SpanBuilderInterface
+    {
+        // Standardized exception attributes
+        $exceptionAttributes = [
+            'exception.message' => $attributes['exception.message'] ?? $exception->getMessage(),
+            'exception.type' => $attributes['exception.type'] ?? get_class($exception),
+            'exception.stacktrace' => $attributes['exception.stacktrace'] ?? $exception->getTraceAsString(),
+            'exception.escaped' => false
+        ];
+
+        // Update span metadata based on exception attributes
+        $this->setAttribute(Tag::ERROR_MSG, $exceptionAttributes['exception.message']);
+        $this->setAttribute(Tag::ERROR_TYPE, $exceptionAttributes['exception.type']);
+        $this->setAttribute(Tag::ERROR_STACK, $exceptionAttributes['exception.stacktrace']);
+
+        // Merge additional attributes
+        $allAttributes = array_merge($exceptionAttributes, iterator_to_array($attributes));
```

Comment:
```suggestion
        $allAttributes = array_merge($exceptionAttributes, \is_array($attributes) ? $attributes : iterator_to_array($attributes));
```
iterator_to_array does not accept arrays in PHP 8.1. Only starting PHP 8.2, but Otel supports PHP 8.1. too.